Core/AppTrackingProtectionAllowlistModel.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  AppTrackingProtectionAllowlistModel.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public class AppTrackingProtectionAllowlistModel {
+    public enum Constants {
+        public static let groupID = "\(Global.groupIdPrefix).apptp"
+        public static let fileName = "appTPallowlist"
+    }
+    
+    private let filename: String
+    
+    lazy private var allowlistUrl: URL? = {
+        let groupContainerUrl = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: Constants.groupID)
+        return groupContainerUrl?.appendingPathComponent(self.filename, conformingTo: .text)
+    }()
+    
+    var allowedDomains: Set<String>
+    
+    public init(filename: String = Constants.fileName) {
```

Comment:
I'd be inclined to make this take the full URL instead of just the filename, but to have a convenience initializer which uses the default name + group container URL, since it would let us provide any URL in the test suite (such as temp directory). What do you think?

Reply:
Actually, I think this is fine in the end, the test suite is already solving the problem I was concerned about